components/sup/src/manager/mod.rs

```diff
@@ -696,10 +705,15 @@ impl Manager {
                 self.shutdown(ShutdownReason::Departed);
                 return Err(sup_error!(Error::Departed));
             }
-            if let Some(SignalEvent::Passthrough(Signal::HUP)) = signals::check_for_signal() {
-                outputln!("Supervisor shutting down for signal");
-                self.shutdown(ShutdownReason::Signal);
-                return Ok(());
+            if env::var("HAB_SUP_IGNORE_SIGNAL").is_err() {
+                match signals::check_for_signal() {
+                    Some(SignalEvent::Passthrough(Signal::HUP)) | Some(SignalEvent::Shutdown) => {
```

Comment:
Shutting down with `ShutdownReason::Signal` will cause the supervisor to restart, which seems at odds with a `SignalEvent::Shutdown` signal.

Reply:
OH! That is not the desired result at all.  Fixing.

Reply:
how would you feel if I added a new `ShutdownReason` called `CancelProcess` that a `shutdown` signal would trigger and result in a butterfly departure?

Reply:
I wouldn't make a new `ShutdownReason`, mainly because that's going away on the branch I'm working on for terminating services from the Supervisor 😄 

I think the `LauncherStopping` reason is what you want here, right?